src/utils.py

```diff
@@ -0,0 +1,77 @@
+import re
+
+from communex.client import CommuneClient
+from substrateinterface import Keypair  # type: ignore
+from substrateinterface import Keypair
+
+IP_REGEX = re.compile(r"\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}:\d+")
+
+
+def set_weights(
+        score_dict: dict[
+            int, float
+        ],  # implemented as a float score from 0 to 1, one being the best
+        # you can implement your custom logic for scoring
+        netuid: int,
+        client: CommuneClient,
+        key: Keypair
+) -> None:
+    """
+    Set weights for miners based on their scores.
+
+    Args:
+        score_dict: A dictionary mapping miner UIDs to their scores.
+        netuid: The network UID.
+        client: The CommuneX client.
+        key: The keypair for signing transactions.
+    """
+
+    # Create a new dictionary to store the weighted scores
+    weighted_scores: dict[int, int] = {}
+
+    # Calculate the sum of all inverted scores
+    scores = sum(score_dict.values())
+    # process the scores into weights of type dict[int, int]
+    # Iterate over the items in the score_dict
```

Comment:
This code has a lot of comments. Function headers are fine but they should be included in every function or not included. Many of the single-line comments are redundant and *should* be evident from the code itself.

I tend to only use single-line comments to include context that cannot be known from the code alone, such as a workaround or a specific choice that was made due to external or non-obvious factors